From cd37423fad3cce884d9dec356373b0e937153bc8 Mon Sep 17 00:00:00 2001
From: Sudheesh Mavila <sudheesh.mavila@amd.com>
Date: Tue, 1 Aug 2017 12:34:44 +0530
Subject: [PATCH 12/32] net: remove __napi_complete() All __napi_complete()
 callers have been converted to use the more standard napi_complete_done(), we
 can now remove this NAPI method for good.

Modified From 02c1602ee7b3e3d062c3eacd374d6a6e3a2ebb73

Signed-off-by: Eric Dumazet <edumazet@google.com>
Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Sudheesh Mavila <sudheesh.mavila@amd.com>
---
 net/core/dev.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)
 mode change 100644 => 100755 net/core/dev.c

diff --git a/net/core/dev.c b/net/core/dev.c
old mode 100644
new mode 100755
index 5e8891b..dd1ca30
--- a/net/core/dev.c
+++ b/net/core/dev.c
@@ -4979,14 +4979,13 @@ bool napi_complete_done(struct napi_struct *n, int work_done)
 		else
 			napi_gro_flush(n, false);
 	}
-	if (likely(list_empty(&n->poll_list))) {
-		WARN_ON_ONCE(!test_and_clear_bit(NAPI_STATE_SCHED, &n->state));
-	} else {
+	if (unlikely(!list_empty(&n->poll_list))) {
 		/* If n->poll_list is not empty, we need to mask irqs */
 		local_irq_save(flags);
-		__napi_complete(n);
+		list_del_init(&n->poll_list);
 		local_irq_restore(flags);
 	}
+	WARN_ON_ONCE(!test_and_clear_bit(NAPI_STATE_SCHED, &n->state));
 	return true;
 }
 EXPORT_SYMBOL(napi_complete_done);
-- 
2.7.4

